Zend/zend_compile.h

```diff
@@ -757,6 +759,16 @@ int zend_add_literal(zend_op_array *op_array, zval *zv);
 #define ZEND_FETCH_CLASS_NO_AUTOLOAD 0x80
 #define ZEND_FETCH_CLASS_SILENT      0x0100
 
+/* reserved names */
+#define ZEND_NOT_RESERVED			0
+#define ZEND_RESERVED_STATIC		1
+#define ZEND_RESERVED_ABSTRACT		2
+#define ZEND_RESERVED_FINAL			3
+#define ZEND_RESERVED_PRIVATE		4
+#define ZEND_RESERVED_PROTECTED		5
+#define ZEND_RESERVED_PUBLIC		6
+#define ZEND_RESERVED_CLASS			7
```

Comment:
I've seen the error message check, but the distinction seems really weird there ... it's pretty much the same message with slightly different wording. That does not seem worthwhile.